src/pybind/mgr/cephadm/services/iscsi.py

```diff
@@ -150,12 +150,24 @@ def post_remove(self, daemon: DaemonDescription) -> None:
         """
         logger.debug(f'Post remove daemon {self.TYPE}.{daemon.daemon_id}')
 
-        # remove config for dashboard iscsi gateways
         ret, out, err = self.mgr.check_mon_command({
-            'prefix': 'dashboard iscsi-gateway-rm',
-            'name': daemon.hostname,
+            'prefix': 'mgr module ls',
+            'format': 'json',
         })
-        logger.info(f'{daemon.hostname} removed from iscsi gateways dashboard config')
+        try:
+            j = json.loads(out)
```

Comment:
```
DEBUG    tests:mgr_module.py:1494 mon_command: 'mgr module ls' -> 0 in 0.000s
ERROR    cephadm.services.iscsi:iscsi.py:161 Failed to parse mgr module ls: Cannot decode JSON: ''
Traceback (most recent call last):
  File "/home/jenkins-build/build/workspace/ceph-pull-requests/src/pybind/mgr/cephadm/services/iscsi.py", line 158, in post_remove
    j = json.loads(out)
  File "/usr/lib/python3.8/json/__init__.py", line 357, in loads
    return _default_decoder.decode(s)
  File "/usr/lib/python3.8/json/decoder.py", line 337, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/usr/lib/python3.8/json/decoder.py", line 355, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
```

seems `ceph mgr module ls` returned an empty str. this is not expected though. might want to reproduce this issue locally.

Reply:
I think the issue is only in the tests as that command isn't mocked

Reply:
```diff
diff --git a/src/pybind/mgr/cephadm/tests/test_cephadm.py b/src/pybind/mgr/cephadm/tests/test_cephadm.py
index a89a72558a..375d17a28c 100644
--- a/src/pybind/mgr/cephadm/tests/test_cephadm.py
+++ b/src/pybind/mgr/cephadm/tests/test_cephadm.py
@@ -920,6 +920,9 @@ spec:
                 placement=ps)
             unmanaged_spec = ServiceSpec.from_json(spec.to_json())
             unmanaged_spec.unmanaged = True
+            # make sure IscsiService.post_remove works
+            cephadm_module._mon_command_mock_mgr_module_ls = lambda *args: json.dumps({
+                                                                                      'enabled_modules': []})
             with with_service(cephadm_module, unmanaged_spec):
 
                 c = cephadm_module.add_daemon(spec)
@@ -1055,6 +1058,9 @@ spec:
     @mock.patch("subprocess.run", mock.MagicMock())
     def test_apply_save(self, spec: ServiceSpec, meth, cephadm_module: CephadmOrchestrator):
         with with_host(cephadm_module, 'test'):
+            # make sure IscsiService.post_remove works
+            cephadm_module._mon_command_mock_mgr_module_ls = lambda *args: json.dumps({
+                                                                                      'enabled_modules': []})
             with with_service(cephadm_module, spec, meth, 'test'):
                 pass
```
should do the trick

Reply:
thanks for the hint